src/main/java/seedu/taassist/logic/commands/AssignCommand.java

```diff
@@ -0,0 +1,98 @@
+
+import seedu.taassist.commons.core.Messages;
+import seedu.taassist.commons.core.index.Index;
+import seedu.taassist.logic.commands.exceptions.CommandException;
+import seedu.taassist.model.Model;
+import seedu.taassist.model.moduleclass.ModuleClass;
+import seedu.taassist.model.student.Student;
+
+/**
+ * Assign students to a class.
+ */
+public class AssignCommand extends Command {
+
+    public static final String COMMAND_WORD = "assign";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Assign students to a class. "
+            + "Parameters: INDEX... (must be positive integers) "
+            + PREFIX_MODULE_CLASS + "CLASS_NAME\n"
+            + "Example: " + COMMAND_WORD + " "
+            + "1 2 3 "
+            + PREFIX_MODULE_CLASS + "CS1231S";
+
+    public static final String MESSAGE_SUCCESS = "Students with %1$s %2$s are assigned to: %3$s";
+    public static final String MESSAGE_DUPLICATE_MODULE_CLASS = "Some students are already assigned with this class";
+
+    private final List<Index> indices;
+    private final ModuleClass moduleClassToAssign;
+
+    /**
+     * Creates an AssignCommand to assign the given {@Code ModuleClass} to students at the given {@Code Indices}.
+     */
+    public AssignCommand(List<Index> indices, ModuleClass moduleClassToAssign) {
+        requireNonNull(indices);
+        requireNonNull(moduleClassToAssign);
+        this.indices = indices;
+        this.moduleClassToAssign = moduleClassToAssign;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (!model.hasModuleClass(moduleClassToAssign)) {
+            throw new CommandException(Messages.MESSAGE_MODULE_CLASS_DOES_NOT_EXIST);
+        }
+
+        List<Student> lastShownList = model.getFilteredStudentList();
+        for (Index index : indices) {
+            if (index.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+            }
+            Student student = lastShownList.get(index.getZeroBased());
+            if (student.getModuleClasses().contains(moduleClassToAssign)) {
+                throw new CommandException(MESSAGE_DUPLICATE_MODULE_CLASS);
+            }
+        }
+        for (Index index : indices) {
+            Student student = lastShownList.get(index.getZeroBased());
+            Set<ModuleClass> moduleClasses = student.getModuleClasses();
+            Set<ModuleClass> newModuleClasses = new HashSet<>();
+            newModuleClasses.addAll(moduleClasses);
+            newModuleClasses.add(moduleClassToAssign);
+
+            Student editedStudent = new Student(
+                    student.getName(),
+                    student.getPhone(),
+                    student.getEmail(),
+                    student.getAddress(),
+                    newModuleClasses);
+            model.setStudent(student, editedStudent);
+        }
```

Comment:
Is it better to let `ParserUtil` handle this? I.e. `AssignCommandParser` calls the `ParserUtil` which parses `1 2 3 4 ...` and returns `List<Students>` containing `Student1`, `Student2`, `Student3` etc.  `AssignCommandParser` then returns `AssignCommand(List<Student>)` instead of `AssignCommand(List<Index>)`

I feel that `AssignCommand` should not know the existence of `Index`

Reply:
That sounds like good abstraction, will make the relevant changes 👍🏻

Reply:
I think `ParserUtil` returning `List<Student>` is fine though; ultimately, the job of a Parser is to take in a string and return whatever it is suppose to represent. But I also agree that returning a list of students from a string of integers may sound strange too.

But if you managed to look for another class to do this it'll be great too!

Reply:
Looking at it again, I feel that `AssignCommandParser` should not know the list of students either... I think I will abstract out the method but let another class handle the part to map indices to students.